modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h

```diff
@@ -53,13 +53,35 @@ class Deserializer;
 /**
  * Manages brushing and linking events for filtering, selecting, and highlighting. When initialized
  * with a BrushingAndLinking inport, changes are propagated using this port if connected.
+ *
+ * Use setInvalidateOn flags if you only want Processor::process to be called for a subset of
+ * brushing actions. Use getModifiedActions if you want to know which BrushingModifications caused a
+ * Processor::process call.
  */
 class IVW_MODULE_BRUSHINGANDLINKING_API BrushingAndLinkingManager : public Serializable {
 public:
-    BrushingAndLinkingManager(BrushingAndLinkingInport* inport,
-                              InvalidationLevel validationLevel = InvalidationLevel::InvalidOutput);
-    BrushingAndLinkingManager(BrushingAndLinkingOutport* outport,
-                              InvalidationLevel validationLevel = InvalidationLevel::InvalidOutput);
+    /**
+     * @param inport owner of the manager.
+     * @param invalidateOn for the modifications that should invalidate the processor
+     * (Processor::process will be called for those).
+     * @param invalidationLevel that will be propagated to the port and thereby upwards in the
+     * network.
+     */
+    BrushingAndLinkingManager(
+        BrushingAndLinkingInport* inport,
+        BrushingModifications invalidateOn = BrushingModifications(flags::any),
```

Comment:
invalidateOn and the invalidationLevel (it should be invalidationLevel not validationLevel) could be combined into one map ` {{{target1,action1}->invalidationLevel1}, {target2,action2}->invalidationLevel2}, ...}`,
maybe one could also support some shortcuts, 
```
{action}->invalidationLevel1} == {all_targets,actions}->invalidationLevel}
{target}->invalidationLevel1} == {target, all_actions}->invalidationLevel}
{invalidationLevel} == {all_targets, all_actions}->invalidationLevel}
```
And if then the order in the list matters, one could for example do:
```
{
    InvalidationLevel::Valid, 
    {BrushingTarget::Row, BrusingModification::Selected} -> InvalidationLevel::InvalidOutput,
    {BrushingTarget::Column, BrusingModification::Filtered} -> InvalidationLevel::InvalidResources
} 
```

Reply:
Fixed in latest commit